src/vsphere_cpi/lib/cloud/vsphere/resources/resource_pool.rb

```diff
@@ -27,12 +28,24 @@ def mob
         if @cluster_config.resource_pool.nil?
           @mob = @root_resource_pool
         else
-          client = @client
-          @mob = client.cloud_searcher.get_managed_object(
-            Vim::ResourcePool,
-            :root => @root_resource_pool,
-            :name => @cluster_config.resource_pool)
-          logger.debug("Found requested resource pool: #{@mob}")
+          begin
+            resource_pool_path_suffix = @cluster_config.resource_pool
+            datacenter_cluster_path_prefix = "#{@datacenter_name}/host/#{@cluster_config.name}/Resources/"
+            full_inventory_path = datacenter_cluster_path_prefix + resource_pool_path_suffix
+            # Replace all multiple consecutive / with single /
+            full_inventory_path.gsub!(/\/+/, '/').chomp!('/')
+            @mob = @client.service_content.search_index.find_by_inventory_path(full_inventory_path)
+            raise "Could not find resource pool #{resource_pool_path_suffix} for inventory path #{full_inventory_path}" if @mob.nil?
+          rescue => e
```

Comment:
I'm not sure I like this pattern:

```ruby
begin
  raise ...
rescue => e
  raise
end
```

I think a simple `if` statement with an early return when found is more appropriate here.

Reply:
Yeah. My original motivation was to implement specific faults to rescue but it is not needed. Will iron this out into `if and elsif and else`